File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.cpp
// Last updated: 4/8/2026, 5:13:44 PM
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int k = 1;
        for (int i = 1; i < nums.size(); i++) {
            if (nums[k - 1] != nums[i]) {
                nums[k] = nums[i];
                k+=1;
            }
        }
        return k;
    }
};

File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.py
# Last updated: 4/8/2026, 5:13:41 PM
class Solution(object):
    def removeDuplicates(self, nums):
        if not nums:
            return 0
        index = 1
        for i in range(1, len(nums)):
            if nums[i] != nums[i - 1]:
                nums[index] = nums[i]
                index += 1
        return index